src/signals/incident/services/map-controls-to-params/index.js

```diff
@@ -39,62 +25,8 @@ const mapControlsToParams = (incident, wizard) => {
     params.incident_date_start = date.format();
   }
 
-  const category_url = incident && incident.subcategory_link ? new URL(incident.subcategory_link).pathname : '';
-  const map = [];
-  let mapMerge = {};
-  forEach(wizard, (step) => {
-    let controls = {};
-    if (step.formFactory && isFunction(step.formFactory)) {
-      const form = step.formFactory(incident);
-      controls = form && form.controls;
-    } else {
-      controls = step.form && step.form.controls;
-    }
-    forEach(controls, (control, name) => {
-      const value = incident[name];
-      const meta = control.meta;
-
-      if (meta && meta.path) {
-        map.push({
-          path: meta.path,
-          value
-        });
-      }
-
-      if (meta && meta.isVisible && meta.pathMerge) {
-        const answer = convertValue(value);
-        if (answer || answer === 0) {
-          mapMerge = {
-            ...mapMerge,
-            [meta.pathMerge]: [
-              ...(mapMerge[meta.pathMerge] || []),
-              {
-                id: name,
-                label: meta.label,
-                category_url,
-                answer
-              }
-            ]
-          };
-        }
-      }
-    });
-  });
-
-  forEach(map, (item) => {
-    let itemValue = convertValue(item.value);
-    if (itemValue || itemValue === 0) {
-      if (isObject(itemValue) && itemValue.id) {
-        itemValue = itemValue.id;
-      }
-
-      set(params, item.path, itemValue);
-    }
-  });
-
-  forEach(mapMerge, (value, key) => {
-    set(params, key, value);
-  });
+  mapValues(params, incident, wizard);
+  mapPaths(params, incident, wizard);
```

Comment:
If these functions don't change the function arguments it will be a lot clearer what is happening.

Reply:
ja dat doen ze wel: ik heb het verbeterd